model/Container/TestQtiServiceProvider.php

```diff
@@ -141,5 +146,15 @@ public function __invoke(ContainerConfigurator $configurator): void
                     service(ToolsStateRepositoryInterface::class),
                 ]
             );
+
+        $services
+            ->set(QtiTestRepositoryInterface::class, QtiTestRepository::class)
```

Comment:
[Optional]
Are use sure it's a good idea to use interface name as a key?

Imagine:
We have multiple places, where we are using this repository.
In the future, we create a new class, which implements this interface.
But we need this new class only in one place (we need to keep previous repository for all other places).

In this case, you need to create a new key (you cannot use interface name, as this will override all the places).

```php
// Service Provider
$services->set(QtiTestRepositoryInterface::class, QtiTestRepository::class);
$services->set(AnotherQtiTestRepository::class, AnotherQtiTestRepository::class);

// Class 1
private function getQtiTestRepository(): QtiTestRepositoryInterface
{
    return $this->getPsrContainer()->get(QtiTestRepositoryInterface::class);
}

// Class 2
private function getQtiTestRepository(): QtiTestRepositoryInterface
{
    return $this->getServiceManager()->getContainer()->get(AnotherQtiTestRepository::class);
}
```

It looks a bit strange, because in first case we use interface as a key and in another - class (or something else, because we cannot use the same key).

But if you will use the same class as a key for both cases, you still can use the interface for type-hint and provide different classes which implements it.

```php
// Service Provider
$services->set(QtiTestRepository::class, QtiTestRepository::class);
$services->set(AnotherQtiTestRepository::class, AnotherQtiTestRepository::class);

// Class 1
private function getQtiTestRepository(): QtiTestRepositoryInterface
{
    return $this->getPsrContainer()->get(QtiTestRepository::class);
}

// Class 2
private function getQtiTestRepository(): QtiTestRepositoryInterface
{
    return $this->getServiceManager()->getContainer()->get(AnotherQtiTestRepository::class);
}
```